packages/web-components/src/components/masthead/top-nav.ts

```diff
@@ -49,7 +51,13 @@ class DDSTopNav extends StableSelectorMixin(HostListenerMixin(BXHeaderNav)) {
   hideDivider = false;
 
   render() {
-    return this._hideNav ? html`` : super.render();
+    return this._hideNav
+      ? html``
+      : html`
+          <nav class="${prefix}--header__nav">
```

Comment:
`<dds-top-nav>` has `.bx--header__nav` style:
https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/v1.12.0/packages/web-components/src/components/masthead/masthead.scss#L128
https://github.com/carbon-design-system/carbon-web-components/blob/v1.7.0/src/components/ui-shell/header.scss#L17

So adding `<nav class="${prefix}--header__nav">` will duplicate the element. Probably we just missed some specific style in `.bx--header__nav`? (e.g. one for specific state)

Reply:
Hm, no I don't think so. That class is missing from the markup.

![image](https://user-images.githubusercontent.com/909118/98937515-7de7f980-24b4-11eb-8c0a-586596491506.png)

The `ul` should have a parent `.bx--header__nav`.

Reply:
> The `ul` should have a parent `.bx--header__nav`. It looks like `<dds-header-nav>` is missing.

@kennylam Thank you for elaborating the thoughts! What I meant is that `<dds-top-nav>` itself should work as `.bx--header__nav`. Here's how the code intends to do that; [`<dds-top-nav>` inherits Carbon core `<bx-header-nav>`](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/v1.12.0/packages/web-components/src/components/masthead/top-nav.ts#L26). [`<bx-header-nav>` has the style of `.bx--header-nav`](https://github.com/carbon-design-system/carbon-web-components/blob/v1.7.0/src/components/ui-shell/header.scss#L16-L17).

Therefore, I think `<dds-top-nav>` tag selector needs more styles defined.

Reply:
Update: From the conversation offline with @ariellalgilmore (thanks Ari!) I now see what’s going on; Seems that we want to prevent the entire `<dds-top-nav>` from being rendered when search box is activated. However, given `render()` only dictates the shadow DOM the host element remains, that causes an extra padding.

@ariellalgilmore You can use `display: contents` style similar to https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/e8af89a06fb41a649529fa441ff29916f4821655/packages/web-components/src/components/carousel/carousel.scss#L16-L19 along with a comment like: https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/e8af89a06fb41a649529fa441ff29916f4821655/packages/web-components/src/components/carousel/carousel.ts#L206

Reply:
Thanks @asudoh! After trying this and discussing further with @kennylam we realized that even adding `display: contents` to `<dds-top-nav>` will not give the correct results. In this case the padding needs to be within `dds-top-nav`, so the search input can go 100% and cover up its own padding. `dds-top-nav` has the padding-left and the red shows that it’s preventing search input to open fully. If the padding was within this element the search would be able to open fully.

I will also change the `<nav>` to a `<div>` as it is not necessary for it to be `<nav>` in this case. 

![image](https://user-images.githubusercontent.com/20210594/99104047-4d2bc100-2595-11eb-8179-b03f2766efd0.png)

Reply:
Thanks @ariellalgilmore for the update! One question; Does React version have such additional elements in addition to `<nav>`?

Reply:
Sent a DM to @ariellalgilmore for further analysis.